docs/docs/howto/registry-authentication.md

```diff
@@ -49,8 +49,170 @@ EOF
+Other registries are allowed, as long as they don't require authentication.
+
+```toml
+[extra-env]
+HTTP_PROXY = "https://proxy.corp"
+HTTPS_PROXY = "https://proxy.corp"
+NO_PROXY = ".corp"
+
+[registries."registry.corp."]
+ca-certs = '''
+-----BEGIN CERTIFICATE-----
+MIIBezCCASGgAwIBAgIUUugBbePTzyVApU4DLSMmHnXXjcwwCgYIKoZIzj0EAwIw
+EzERMA8GA1UEAwwIWW91ck5hbWUwHhcNMjUxMDIxMTUwNDI0WhcNMjYxMDIxMTUw
+NDI0WjATMREwDwYDVQQDDAhZb3VyTmFtZTBZMBMGByqGSM49AgEGCCqGSM49AwEH
+A0IABIgsA5IEeiBq6jDpH2ttxrI96beeOqa+EpGqmznQmzpFkPEpLWMUt21Ien71
+rxdeFC7ySuuu95VPjSvO7EUM9qyjUzBRMB0GA1UdDgQWBBTVnuI2o36Mrja3RvwE
+82lWg2m19zAfBgNVHSMEGDAWgBTVnuI2o36Mrja3RvwE82lWg2m19zAPBgNVHRMB
+Af8EBTADAQH/MAoGCCqGSM49BAMCA0gAMEUCIGmEkl8jxjxqyAxs3QoAXeIx++Bz
+Zm9dwbeTbrKysrGXAiEA8ce6iyJUCZCZVVJs/HDLcPbOKc2EPZvdcGGjIlGXulo=
+-----END CERTIFICATE-----
+'''
+
+[registries."ghcr.io."]
+auth = "YnVyZ2VyZGV2OnRoaXNpc25vdG15cGFzc3dvcmQ="
+```
+
+##### Example 2
+
+In this scenario, all container images are served from a public registry.
+However, the image owner wants to make sure that the traffic can't be intercepted by rogue CAs.
+Other registries are strictly forbidden.
+
+```toml
+[registries."very-secure.registri.es."]
+auth = "bmljZTp0cnk="
+ca-certs = '''
+Root CA 1
+-----BEGIN CERTIFICATE-----
+MIIBfDCCASGgAwIBAgIUU5G42y9bIh8+AU38qVOmKocc0CwwCgYIKoZIzj0EAwIw
+EzERMA8GA1UEAwwIWW91ck5hbWUwHhcNMjUxMDIxMTUyNzEwWhcNMjYxMDIxMTUy
+NzEwWjATMREwDwYDVQQDDAhZb3VyTmFtZTBZMBMGByqGSM49AgEGCCqGSM49AwEH
+A0IABOJlyBb/sHBmHRncTqk4lm6hBkBYlZGcScXfl/IuAVVIo4zCGBzCmvc7jYc2
++gyVp+wxuvm7NRza4e1QOfJfrxOjUzBRMB0GA1UdDgQWBBTRE8qju+GIWzr5xCik
+MdBJFOd1lzAfBgNVHSMEGDAWgBTRE8qju+GIWzr5xCikMdBJFOd1lzAPBgNVHRMB
+Af8EBTADAQH/MAoGCCqGSM49BAMCA0kAMEYCIQCn+fVmAzB8HOakKGLx6oXF0WP0
+GJibphhjfHPdNWEDdQIhAN3KFNWIYtE35+/rZb5I+oVKnqKS8igdIU9lXmpOps1j
+-----END CERTIFICATE-----
+Root CA 2
+-----BEGIN CERTIFICATE-----
+MIIBezCCASGgAwIBAgIUUugBbePTzyVApU4DLSMmHnXXjcwwCgYIKoZIzj0EAwIw
+EzERMA8GA1UEAwwIWW91ck5hbWUwHhcNMjUxMDIxMTUwNDI0WhcNMjYxMDIxMTUw
+NDI0WjATMREwDwYDVQQDDAhZb3VyTmFtZTBZMBMGByqGSM49AgEGCCqGSM49AwEH
+A0IABIgsA5IEeiBq6jDpH2ttxrI96beeOqa+EpGqmznQmzpFkPEpLWMUt21Ien71
+rxdeFC7ySuuu95VPjSvO7EUM9qyjUzBRMB0GA1UdDgQWBBTVnuI2o36Mrja3RvwE
+82lWg2m19zAfBgNVHSMEGDAWgBTVnuI2o36Mrja3RvwE82lWg2m19zAPBgNVHRMB
+Af8EBTADAQH/MAoGCCqGSM49BAMCA0gAMEUCIGmEkl8jxjxqyAxs3QoAXeIx++Bz
+Zm9dwbeTbrKysrGXAiEA8ce6iyJUCZCZVVJs/HDLcPbOKc2EPZvdcGGjIlGXulo=
+-----END CERTIFICATE-----
+'''
+
+[registries."."]
+ca-certs = "no PEM here means no CA certificates"
+```
+
+##### Example 3
+
+In this scenario, there's an HTTP-only registry deployed into the cluster for ease of use.
+Transport security for this internal registry isn't important to the operators.
+Other registries should be used anonymously, but with TLS.
+
+```toml
+[registries."registry.default.svc.cluster.local."]
+insecure-skip-verify = true
+```
+
+##### Example 4
+
+This scenario is equivalent to the prior behavior: web PKI and no authentication.
+
+```toml
+```
```

Comment:
We should frame this along the lines of "if you don't add a secret, this happens".